changelogs/unreleased/8573-add-support-for-annotated-types.yml

```diff
@@ -0,0 +1,4 @@
+description: Add support for annotated types to plugins.
+issue-nr: 8573
+change-type: minor
+destination-branches: [master, iso8]
```

Comment:
We probably want to include this in iso7 as well. Discussion still pending, see [Slack](https://inmanta.slack.com/archives/CKRF0C8R3/p1737472131458259).